### PR TITLE
dark sky / freezing weather / sync

### LINKED ISSUE #5: Weather Underground API is sunsetting 2019-02-15, need to evaluate consequences for Freezing Saddles weather integrations
It looks like the Weather Underground API is sunsetting at the end of 2018, making it useless for the 2019 competition, alas: https://apicommunity.wunderground.com/weatherapi/topics/end-of-service-for-the-weather-underground-api

I haven't looked deep enough to see whether the https://github.com/freezingsaddles/freezing-sync/tree/master/freezing/sync/wx/ncdc code would offer enough weather information to be usable for this app yet.

Other alternatives that look promising are:
* https://darksky.net/dev
*  https://openweathermap.org/api

### LINKED ISSUE #15: Rework weather integration to use Dark Sky APIs
Instead of using Weather Underground (see issue #5) let's use the Dark Sky APIs instead: https://darksky.net/dev

### PR DESCRIPTION
**Pull request by merlinorg**
Caching dark sky weather sync. The API limit is 1000 requests/day at the free tier which, with an 80% cache hit rate, ought to give us about 5000 rides a day. I'm not thrilled with the precipitation data, the docs make it sound a bit dubious, but it seems good enough. Plausibly resolves #15, closes #5.

@spfrantz i hope you don't mind my kicking this.

### CODE REVIEW

**Comment by merlinorg** [comment]
Plausibly closes #15.

**Comment by obscurerichard** [comment]
We can test this without a cache, but I'd really like to make sure we have a cache in place before we take it to production.

**Comment by merlinorg** [comment]
As is thy will, @obscurerichard, so cached it be.

**Comment by merlinorg** [comment]
The dark sky library I originally used was GPL and thus incompatible with FS. The others I found proved unusable so we now have a minimal local model. I ran this on some 2019 rides and compared with the original weather data in the database and the results seemed not too dissimilar.

File: freezing/sync/data/weather.py [hunk 1]
```diff
@@ -70,51 +72,59 @@ def sync_weather(self, clear:bool = False, limit: int = None, cache_only: bool =
...
-                rw.ride_temp_avg = avg([o.temp for o in ride_observations])
+                rw.ride_windchill_start = start_obs.apparent_temperature
+                rw.ride_windchill_end = end_obs.apparent_temperature
+                rw.ride_windchill_avg = mean([o.apparent_temperature for o in ride_observations])
 
-                rw.ride_windchill_start = start_obs.windchill
-                rw.ride_windchill_end = end_obs.windchill
-                rw.ride_windchill_avg = avg([o.windchill for o in ride_observations])
+                # scale the cumulative precipitation over the observation period by the fraction of time spent moving
+                scale = timedelta(hours=len(ride_observations)).total_seconds() / ride.moving_time
```
Comment (merlinorg) [inline freezing/sync/data/weather.py]: This is a change. Formerly, if you rode five minutes in a snowstorm you would be credited with 2 to 3 times one hour's precipitation. In the same scenario now, you would get credit for 5/60 of that hour's precipitation.

**Pull request merged**